examples/esp_idf/lightdb_stream/main/app_main.c

```diff
@@ -0,0 +1,192 @@
+    if (err)
+    {
+        GLTH_LOGW(TAG, "Failed to push temperature: %d", err);
+        return;
+    }
+}
+
+static void temperature_push_sync(const float *temp)
+{
+    char sbuf[sizeof("{\"temp\":-4294967295.123456}")];
+    int err;
+
+    snprintf(sbuf, sizeof(sbuf), "{\"temp\":%f}", *temp);
+
+    err = golioth_stream_set_sync(client,
+                                  "sensor",
+                                  GOLIOTH_CONTENT_TYPE_JSON,
+                                  (uint8_t *) sbuf,
+                                  strlen(sbuf),
+                                  1);
+
+    if (err)
+    {
+        GLTH_LOGW(TAG, "Failed to push temperature: %d", err);
+        return;
+    }
+
+    GLTH_LOGI(TAG, "Temperature successfully pushed");
+}
+
+void app_main(void)
+{
+    float temp;
+    int err;
+
+    GLTH_LOGI(TAG, "Start LightDB Stream sample");
+
+    // Initialize NVS first. For this example, it is assumed that WiFi and Golioth
+    // PSK credentials are stored in NVS.
+    nvs_init();
+
+    // Create a background shell/CLI task (type "help" to see a list of supported commands)
+    shell_start();
+
+    // If the credentials haven't been set in NVS, we will wait here for the user
+    // to input them via the shell.
+    if (!nvs_credentials_are_set())
+    {
+        while (1)
+        {
+            vTaskDelay(1000 / portTICK_PERIOD_MS);
+            GLTH_LOGW(TAG, "WiFi and golioth credentials are not set");
+            GLTH_LOGW(TAG, "Use the shell settings commands to set them, then restart");
+            vTaskDelay(UINT32_MAX);
+        }
+    }
+
+    // Initialize WiFi and wait for it to connect
+    wifi_init(nvs_read_wifi_ssid(), nvs_read_wifi_password());
+    wifi_wait_for_connected();
+
+    // Now we are ready to connect to the Golioth cloud.
+    const char *psk_id = nvs_read_golioth_psk_id();
+    const char *psk = nvs_read_golioth_psk();
+
+    struct golioth_client_config config = {
+        .credentials =
+            {
+                .auth_type = GOLIOTH_TLS_AUTH_TYPE_PSK,
+                .psk =
+                    {
+                        .psk_id = psk_id,
+                        .psk_id_len = strlen(psk_id),
+                        .psk = psk,
+                        .psk_len = strlen(psk),
+                    },
+            },
+    };
+    client = golioth_client_create(&config);
+    assert(client);
+
+    _connected_sem = xSemaphoreCreateBinary();
+
+    golioth_client_register_event_callback(client, on_client_event, NULL);
+
+    GLTH_LOGW(TAG, "Waiting for connection to Golioth...");
+    xSemaphoreTake(_connected_sem, portMAX_DELAY);
+
+    while (true)
+    {
+        /* Synchronous using JSON object */
+        err = get_temperature(&temp);
+        if (err)
+        {
+            vTaskDelay(1000 / portTICK_PERIOD_MS);
+            continue;
+        }
+
+        GLTH_LOGI(TAG, "Sending temperature %f", temp);
+
+        temperature_push_sync(&temp);
```

Comment:
Is there any reason to pass this by reference instead of by value?

Reply:
No reason. This was how the zephyr example did it and I followed the same method. I'll change it.